sdks/java/io/kafka/upgrade/src/main/java/org/apache/beam/sdk/io/kafka/upgrade/KafkaIOTranslation.java

```diff
@@ -245,7 +245,7 @@ public Row toConfigRow(Read<?, ?> transform) {
           transform = transform.withTopics(new ArrayList<>(topics));
         }
         Collection<Row> topicPartitionRows = configRow.getArray("topic_partitions");
-        if (topicPartitionRows != null) {
+        if (topicPartitionRows != null && !topicPartitionRows.isEmpty()) {
```

Comment:
same

Reply:
I couldn't find one or add one that also satisfies that checker-framework nullness check. So leaving as is.